flake.nix

```
@@ -0,0 +1,52 @@
+{
+  description = "Rustlings devShell.";
+
+  inputs = {
+    flake-compat = {
+      url = "github:edolstra/flake-compat";
+      flake = false;
+    };
+    flake-utils.url = "github:numtide/flake-utils";
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+  };
+
+  outputs = { self, flake-compat, flake-utils, nixpkgs }:
```

`flake-compat` isn't used here, so IMO it's cleaner to not reference it? Up to you, though.

```suggestion
  outputs = { self, flake-utils, nixpkgs, ... }:
```

---

Good call!